### PR TITLE
Rover: stop using distance_cm method on rangefinder

### PR DESCRIPTION
**Pull request by peterbarker**
.... also, add autotest for same.  This involved moving the Proximity-RangeFinder posts code up into SITL where we could do the ray-casting for rangefinders.

### CODE REVIEW

File: libraries/SITL/SIM_Aircraft.cpp [hunk 1]
```diff
@@ -483,11 +483,19 @@ void Aircraft::fill_fdm(struct sitl_fdm &fdm)
 // is bouncing off:
 float Aircraft::perpendicular_distance_to_rangefinder_surface() const
 {
-    return sitl->height_agl;
+    switch ((Rotation)sitl->sonar_rot.get()) {
+    case Rotation::ROTATION_PITCH_270:
+        return sitl->height_agl;
+    case ROTATION_NONE ... ROTATION_YAW_315:
```
Comment (khancyr) [inline libraries/SITL/SIM_Aircraft.cpp:489]: how is this valid ?

**Review by khancyr — APPROVED** [review]
looks good

NFC for rover

File: libraries/SITL/SIM_Aircraft.cpp (hunk 1 above)
Comment (peterbarker) [reply libraries/SITL/SIM_Aircraft.cpp:489]: Magic!

Those are kept consecutive in our rotations list so we can get away with doing this....

The ellipsis is just "all enumeration entries between the two endpoints...

**Review by IamPete1 — APPROVED** [review]
Rover changes look good, SITL is only SITL....

**Comment by rmackay9** [comment]
I agree the Rover part of the change looks fine.

**Review by rmackay9 — APPROVED** [review]

**Pull request merged**